src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -249,7 +253,7 @@ private static PoisonedFileEntry CheckSingleFile(IEnumerable<CatalogPackageEntry
             }
 
             var poisonEntry = new PoisonedFileEntry();
-            poisonEntry.Path = Utility.MakeRelativePath(fileToCheck, rootPath);
+            poisonEntry.Path = Utility.MakeRelativePath(candidate.FullPath, candidate.RootPath);
```

Comment:
I was expecting the call to MakeRelativePath to be removed.  Isn't poisonEntry.Path what we wanted to be the full path?  Can you share a sample poison report with these changes?

Reply:
The full path includes my personal root directories in this instance. So for example, the full path on my computer was `/root/vmr/artifacts/x64/Release`. That is why it is still necessary to remove this "root."

In hindsight, I can remove this root path from the full path when the candidates queue is first initialized, eliminating both the RootPath parameter and the call to Utility.MakeRelativePath() in this particular function.

Reply:
XML files aren't supported for upload so I can't upload the poison-usage.xml, but here are some lines from a report produced with these changes:

```
<File Path="dotnet-sdk-8.0.100-preview.7.23376.1-fedora.38-x64.tar.gz/packs/NETStandard.Library.Ref/2.1.0/ref/netstandard2.1/Microsoft.Win32.Primitives.dll">
    <Hash>3c20108b9bd1fd574150473b1ad8f1117db325b5bacb2b4c411fbcbb5e5e8196</Hash>
    <Type>AssemblyAttribute</Type>
  </File>
  <File Path="dotnet-sdk-8.0.100-preview.7.23376.1-fedora.38-x64.tar.gz/packs/NETStandard.Library.Ref/2.1.0/ref/netstandard2.1/mscorlib.dll">
    <Hash>2061426f79957ed81c2ec8cfc700f874e5ac4999e47db695f83d1c21e39b32e7</Hash>
    <Type>AssemblyAttribute</Type>
  </File>
  <File Path="Private.SourceBuilt.Artifacts.8.0.100-preview.7.23376.1.fedora.38-x64.tar.gz/dotnet-format.8.0.435601.nupkg/tools/net8.0/any/Microsoft.Extensions.Logging.Abstractions.dll">
    <Hash>8caf1cb51ee7e9b8ca5c26f080961036793152af78fa7a14a2d3a39ed78e967c</Hash>
    <Type>AssemblyAttribute</Type>
  </File>
  <File Path="Private.SourceBuilt.Artifacts.8.0.100-preview.7.23376.1.fedora.38-x64.tar.gz/dotnet-format.8.0.435601.nupkg/tools/net8.0/any/Microsoft.Extensions.Logging.dll">
    <Hash>bd02eb6b48c7765ae1fbab483c8905a7b311155b0b4a490b75a477202c61b42d</Hash>
    <Type>AssemblyAttribute</Type>
  </File>
```

Reply:
Removed RootPath member and removed call to MakeRelativePath in `CheckSingleFile`: https://github.com/dotnet/installer/pull/17438/commits/b99b930bf2726d24488910226e0fac3f3575e515

Reply:
The trick to uploading unsupported file types is to wrap them in a zip file and upload the zip.

Reply:
> The full path includes my personal root directories in this instance. So for example, the full path on my computer was /root/vmr/artifacts/x64/Release/zipfile.tar/..... Thus, it was necessary to remove /root/vmr/artifacts/x64/Release/ (the root) from the full path.

I wonder if this is still going to cause some confusion?  Do you think it would help to make the path relative to the VMR root?

Reply:
IMO, having the root as the tarball is enough information to me. I wonder that if this code is run on AzDo pipelines that including the path relative to the VMR would cause more confusion than starting the path with the zip file.

Reply:
The reason I mention this is that I don't think everyone knows where the tarballs are located.  Seeing artifacts/<arch>/<flavor>/tarball makes it obvious.  The paths will be the same regardless of where the build occurs.

Reply:
I see. I will add that in!

Reply:
Updated in this commit: https://github.com/dotnet/installer/pull/17438/commits/d2355d93809fcf6e01105f6df93f02bc2faf3319